envisage/tests/test_extension_registry_mixin.py

```diff
@@ -0,0 +1,117 @@
+
+        # Make sure there are no extensions.
+        extensions = registry.get_extensions("my.ep")
+        self.assertEqual(0, len(extensions))
+
+        # Make sure there are no extension points.
+        extension_points = registry.get_extension_points()
+        self.assertEqual(0, len(extension_points))
+
+    def test_add_extension_point(self):
+        """ add extension point """
+
+        registry = self.registry
+
+        # Add an extension *point*.
+        registry.add_extension_point(self._create_extension_point("my.ep"))
+
+        # Make sure there's NO extensions.
+        extensions = registry.get_extensions("my.ep")
+        self.assertEqual(0, len(extensions))
+
+        # Make sure there's one and only one extension point.
+        extension_points = registry.get_extension_points()
+        self.assertEqual(1, len(extension_points))
+        self.assertEqual("my.ep", extension_points[0].id)
+
+    def test_get_extension_point(self):
+        """ get extension point """
+
+        registry = self.registry
+
+        # Add an extension *point*.
+        registry.add_extension_point(self._create_extension_point("my.ep"))
+
+        # Make sure we can get it.
+        extension_point = registry.get_extension_point("my.ep")
+        self.assertNotEqual(None, extension_point)
+        self.assertEqual("my.ep", extension_point.id)
+
+    def test_remove_empty_extension_point(self):
+        """ remove empty_extension point """
+
+        registry = self.registry
+
+        # Add an extension point...
+        registry.add_extension_point(self._create_extension_point("my.ep"))
+
+        # ...and remove it!
+        registry.remove_extension_point("my.ep")
+
+        # Make sure there are no extension points.
+        extension_points = registry.get_extension_points()
+        self.assertEqual(0, len(extension_points))
+
+    def test_remove_non_existent_extension_point(self):
+        """ remove non existent extension point """
+
+        registry = self.registry
+
+        with self.assertRaises(UnknownExtensionPoint):
+            registry.remove_extension_point("my.ep")
+
+    def test_remove_non_existent_listener(self):
+        """ remove non existent listener """
+
+        registry = self.registry
+
+        def listener(registry, extension_point, added, removed, index):
+            """ Called when an extension point has changed. """
+
+            self.listener_called = (registry, extension_point, added, removed)
+
+        with self.assertRaises(ValueError):
+            registry.remove_extension_point_listener(listener)
+
+    ###########################################################################
+    # Private interface.
+    ###########################################################################
+
+    def _create_extension_point(self, id, trait_type=List, desc=""):
+        """ Create an extension point. """
+
+        return ExtensionPoint(id=id, trait_type=trait_type, desc=desc)
```

Comment:
The `ExtensionRegistryTestCase` overrides this `_create_extension_point` method, so this is no longer private. Would it make sense to make it public and remove the one on `ExtensionRegistryTestCase`?

Another approach to consider: this might be better extracted to a function that does not attach to a class. Then one can simply import the function to reuse it without having to inherit a mixin that also brings in additional tests. Note that this function is also defined in `test_extension_point_binding` which is not using this mixin.

Reply:
For now I went with the first approach.  I noticed the function is defined in `test_extension_point` as well.  It might make sense that that is where the public function should sit?

seeing `from envisage.tests.test_extension_point import create_extension_point` might read a little better than `from envisage.tests.test_extension_registry_mixin import create_extension_point`?  I think for now I will leave that to a separate PR, should I open an issue?

Reply:
> For now I went with the first approach. I noticed the function is defined in test_extension_point as well. It might make sense that that is where the public function should sit?

Sounds good to me. Thank you!

> seeing from envisage.tests.test_extension_point import create_extension_point might read a little better than from envisage.tests.test_extension_registry_mixin import create_extension_point? I think for now I will leave that to a separate PR, should I open an issue?

Having a method here is fine. There is only one another `create_extension_point`, and the function is very basic. It is not like we have many identically complicated functions where we have to spend a lot of efforts maintaining, so to me, it is not really an issue.